File: src/limelight/ui/model/inputs/Button.java
package limelight.ui.model.inputs;

import limelight.ui.model.updates.Updates;

import javax.swing.*;
import java.awt.*;

public class Button extends JButton
{
  private ButtonPanel panel;

  public Button(ButtonPanel panel)
  {
    this.panel = panel;
  }

  public void repaint()
  {
    if(panel != null)
      panel.setNeededUpdate(Updates.shallowPaintUpdate);
  }

  public void repaint(long tm, int x, int y, int width, int height)
  {
    if(panel != null)
      panel.setNeededUpdate(Updates.shallowPaintUpdate);
  }

  public void repaint(Rectangle r)
  {
    if(panel != null)
      panel.setNeededUpdate(Updates.shallowPaintUpdate);
  }

  public boolean isShowing()
  {
    return true;
  }
}

File: lib/limelight/players/button.rb
#- Copyright 2008 8th Light, Inc.
#- Limelight and all included source files are distributed under terms of the GNU LGPL.

module Limelight
  module Players
    
    module Button
      class << self

        def extended(prop)
          button = Limelight::UI::Model::Inputs::ButtonPanel.new
          prop.panel.add(button)
          prop.button = button.button
          set_default_styles(prop)
        end
        
        def set_default_styles(prop)
          prop.style.width = "100"
          prop.style.height = "29"
        end
      end
  
      attr_accessor :button
      
    end
  
  end
end